Assets/Scripts/Strategy/BaseManagement/IBaseManagementState.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public interface IBaseManagementState : IUpdatable
+{
+    void SelectBuildingTier();
+
+    void SelectBuilding();
+
+    void PlaceBuilding();
```

Comment:
Wouldn't it make sense for these to have arguments?

Reply:
Yes, potentially. I will consider args in restructure of state system.